replicatedhq/ship
# Write out kustomize overlays

What I Did
------------

After user finalizes kustomization step, write out overlays to the filesystem.

How I Did it
------------

- move state behind an interface so we can mock it, add mockgen steps (interface is `Manager` old struct is called `MManager`, I would make the struct private but too many tests grab for it directly)
- needed to tweak `specs.Resolver`/`ResolverTest` a little, but this should eventually use the mock instead
- When user saves kustomize, load overlays from state, write each patch file, tracking each one and adding to `kustomization.yml` as well
- add some basic testing for `lifecycle/kustomize#kustomizer`
- rename `state.Overlay#Files` to `Patches`

How to verify it
------------

Run `ship init` flow, see kustomize overlay written out

Description for the Changelog
------------

Kustomize step writes overlays to output directory

![](http://cdn.earthporm.com/wp-content/uploads/2014/06/sinking-boat-92.jpg)
